addons/mass_mailing/models/mass_mailing.py

```diff
@@ -578,7 +578,25 @@ def _get_blacklist(self):
         blacklist = {}
         target = self.env[self.mailing_model_real]
         mail_field = 'email' if 'email' in target._fields else 'email_from'
-        if 'opt_out' in target._fields:
+        if self.mailing_model_real == "mail.mass_mailing.contact":
+            # If user is opt_out on One list but not on another or if two user
+            # with same email address, one opted in and the other one
+            # opted out, send the mail anyway
+            query = """
+                SELECT lower(substring(email, '([^ ,;<@]+@[^> ,;]+)')), opt_out
+                FROM mail_mass_mailing_contact
+                WHERE id IN (
+                    SELECT contact_id
+                    FROM mail_mass_mailing_contact_list_rel
+                    WHERE list_id IN %s)
+                    AND substring(email, '([^ ,;<@]+@[^> ,;]+)') IS NOT NULL;
+            """
+            self._cr.execute(query, (tuple(self.contact_list_ids.ids),))
+            contacts = self._cr.fetchall()
+            opt_out_contacts = [c[0] for c in contacts if c[1]]
```

Comment:
Apply these 3 suggestions to make it easier to read and faster:

```suggestion
            opt_out_contacts = {c[0] for c in contacts if c[1]]}
```